lib/full-site-editing/block-templates.php

```diff
@@ -105,6 +105,42 @@ function _gutenberg_get_template_files( $template_type ) {
 	return $template_files;
 }
 
+/**
+ * Parses wp_template content and injects the current theme's
+ * stylesheet as a theme attribute into each wp_template_part
+ *
+ * @param string $template_content serialized wp_template content.
+ * @param string $theme the active theme's stylesheet.
+ *
+ * @return string Updated wp_template content.
+ */
+function _inject_theme_attribute_in_content( $template_content, $theme ) {
+	$has_updated_content = false;
+	$new_content         = '';
+	$template_blocks     = parse_blocks( $template_content );
+
+	foreach ( $template_blocks as $key => $block ) {
+		if (
+			'core/template-part' === $block['blockName'] &&
+			! isset( $block['attrs']['theme'] ) &&
+			wp_get_theme()->get_stylesheet() === $theme
```

Comment:
Would it make sense to extract this as a bail condition before the for loop? It seems to be independent of any template blocks data being looped over.

Going a bit further, based on this description
```@param string $theme the active theme's stylesheet.```

Can we drop this parameter from the function, given that we can obtain this with `wp_get_theme()->get_stylesheet()`? 🤔 It looks to me that if the passed `$theme` parameter that is different from `wp_get_theme()->get_stylesheet()` this function will essentially be a no-op.

Reply:
> Would it make sense to extract this as a bail condition before the for loop? It seems to be independent of any template blocks data being looped over.

If we keep `$theme` as a param, this makes sense.

> Can we drop this parameter from the function, given that we can obtain this with wp_get_theme()->get_stylesheet()? 🤔 It looks to me that if the passed $theme parameter that is different from wp_get_theme()->get_stylesheet() this function will essentially be a no-op.

I mentioned something similar here https://github.com/WordPress/gutenberg/pull/28088#discussion_r556328434. It didn't generate a lot of discussion, but I prefer what you're describing.